src/tribler-core/tribler_core/components/ipv8/ipv8_component.py

```diff
@@ -30,6 +32,9 @@ class Ipv8Component(RestfulComponent):
     _peer_discovery_community: Optional[DiscoveryCommunity] = None
 
     async def run(self):
+        self.ipv8 = None
+        self.peer = None
+
         await super().run()
```

Comment:
**nit** from my POV it is safer (in a long term) to keep an inheritance chain as simple and as predictable as possible. Therefore I suggest always call the overridden `super()` method first, and any other operations second.
```suggestion
    async def run(self):
        await super().run()
        self.ipv8 = None
        self.peer = None


```